cpmpy/transformations/flatten_model.py

```diff
@@ -163,7 +163,7 @@ def flatten_constraint(expr):
                 if expr.args[1].name == 'and':
                     a1s = expr.args[1].args
                     a0 = expr.args[0]
-                    newlist.extend(flatten_constraint([a0.implies(a1) for a1 in a1s]))
+                    lst_of_expr.extend([a0.implies(a1) for a1 in a1s])
```

Comment:
can you take this 'modify list you are iterating over' one out? its in a separate PR